### ISSUE TITLE
FolderResponse handling from move_envelopes() raises exception

### ISSUE THREAD

**Issue by gregality** [docusign/docusign-python-client#18]
Calling move_envelopes() from the FoldersApi successfully moves the envelopes as desired, but the response generates the exception and traceback below.  

The data attribute of the response (FolderResponse) is empty, which I think is okay, but because it is data=b'' instead of data=None, __deserialize_model() (api_client.py) tries to handle it and raises the exception below.  

I've also included a screenshot of my debugger where the data attribute is initially populated (to demonstrate that it is coming from the API like this). 

```
  File "ve/app/scripts/main.py", line 116, in process_documents
    data = folders_api.move_envelopes(account_id, processed_folder_id, folders_request=folders_request)
  File "ve/lib/python3.5/site-packages/docusign_esign/apis/folders_api.py", line 333, in move_envelopes
    (data) = self.move_envelopes_with_http_info(account_id, folder_id, **kwargs)
  File "ve/lib/python3.5/site-packages/docusign_esign/apis/folders_api.py", line 420, in move_envelopes_with_http_info
    collection_formats=collection_formats)
  File "ve/lib/python3.5/site-packages/docusign_esign/api_client.py", line 385, in call_api
    _return_http_data_only, collection_formats, _preload_content, _request_timeout)
  File "ve/lib/python3.5/site-packages/docusign_esign/api_client.py", line 216, in __call_api
    return_data = self.deserialize(response_data, response_type)
  File "ve/lib/python3.5/site-packages/docusign_esign/api_client.py", line 298, in deserialize
    return self.__deserialize(data, response_type)
  File "ve/lib/python3.5/site-packages/docusign_esign/api_client.py", line 338, in __deserialize
    return self.__deserialize_model(data, klass)
  File "ve/lib/python3.5/site-packages/docusign_esign/api_client.py", line 686, in __deserialize_model
    and instance.attribute_map[attr] in data \
TypeError: a bytes-like object is required, not 'str'
```

![image](https://user-images.githubusercontent.com/12747583/45827314-27137a80-bcc4-11e8-92b8-b55e07cd06fe.png)

**Comment by gregality** [comment]
Thank you!  Just to confirm... your assessment is that the correct solution is to make a change to the API vs. an update to the Python library?

**Comment by LarryKlugerDS** [comment]
The error code is from the underlying API itself. The SDK, at this time, is a pretty lightweight wrapper around the API calls. The SDK handles serialization / deserialization but that's about it. Since the API payload is JSON, you can do the serialization / deserialization yourself without the SDK if you wish.

For error responses, I suggest that you also look at the full body of the response. It is available:

````
except docusign.rest.ApiException as err:
    body = err.body.decode('utf8')
    jsonErr = json.loads(body)
````

In the future, we plan an additional "business logic" layer. It will assist, for example, in managing the access token for you (tracking expiration, automatically renewing when possible, etc.).

**Comment by gregality** [comment]
I'm not getting an ApiException.  I'm getting a TypeError.  It isn't clear to me whether the API  is returning a bad type/value or if the SDK should be handling it.

**Comment by gregality** [comment]
Got it.  Thanks!

**Comment by mmallis87** [comment]
@gregality please upgrade to version 2.0.0 and the issue should be gone. In case you're looking for an example, you can refer to the new test case in test/unit_tests.py.

**Comment by LarryKlugerDS** [comment]
This bug was fixed in v2.0.0 of the Python SDK. I'm closing this issue, please comment on it if there are further related issues.
Thank you.
